tests/cpp/test_multidevice_transformer.cpp

```diff
@@ -984,6 +998,275 @@ TEST_P(DistributedTransformerTest, Forward) {
   validate(expected_outputs, outputs);
 }
 
+TEST_P(DistributedTransformerTest, Backward) {
+  auto dtype = GetParam();
+  at::ScalarType at_dtype = data_type_to_aten(dtype);
+  auto fusion = std::make_unique<Fusion>();
+  FusionGuard fg(fusion.get());
+  const auto mesh = DeviceMesh::createForNumDevices(D);
+  constexpr float kEps = 1e-5;
+  auto eps = IrBuilder::create<Val>(kEps);
+  std::vector<int64_t> norm_shape{E}; // 768
+
+  // Note: Deviate from the thunder trace with layer norm recomputation
+  // Thunder saves intermediate values that are not exposed by ATen
```

Comment:
> not exposed by ATen

Can you be more specific about this? ATen should have enough low-level APIs to rebuild the Thunder trace. I guess you meant to say these intermediate values are not exposed by certain ATen APIs, making it hard to build the reference implementation. Correct?

Reply:
Disregard! The ATen implementation actually does expose the intermediates we need, they were just named slightly different form nvfuser! Will fix.